src/coreclr/jit/valuenum.cpp

```diff
@@ -8918,16 +8918,16 @@ void Compiler::fgValueNumberHWIntrinsic(GenTree* tree)
     assert(hwIntrinsicNode != nullptr);
 
     // For safety/correctness we must mutate the global heap valuenumber
-    //  for any HW intrinsic that performs a memory store operation
+    // for any HW intrinsic that performs a memory store operation
     if (hwIntrinsicNode->OperIsMemoryStore())
     {
         fgMutateGcHeap(tree DEBUGARG("HWIntrinsic - MemoryStore"));
     }
 
-    // Check for any intrintics that have variable number of args or more than 2 args
-    // For now we will generate a unique value number for these cases.
-    //
-    if ((HWIntrinsicInfo::lookupNumArgs(hwIntrinsicNode->gtHWIntrinsicId) == -1) ||
+    const bool isVariableArgs = HWIntrinsicInfo::lookupNumArgs(hwIntrinsicNode->gtHWIntrinsicId) == -1;
+
+    // In case of functions with variable number of args we only support single-arg ones.
```

Comment:
Just wondering, why 0 or 1 and not 0, 1, or 2?

Reply:
Variable length args only impact `Vector.Create` calls for ARM, but also impact cases like `Sse.ReciprocalScalar` for x86, where the upper limit is 2 args.

Reply:
Initially I wanted to add support for args >= 2 cases later but since you noticed this case I've added it as part of this pr.

Repro:
```csharp
static int Test()
{
    return Sse2.MoveMask(Sse2.Add(
                Vector128.Create(1, 1), 
                Vector128.Create(1, 1)).AsByte());
}
```
New codegen:
```asm
; Method Program:Test():int
G_M58954_IG01:
       vzeroupper 
						;; bbWeight=1    PerfScore 1.00

G_M58954_IG02:
       vmovupd  xmm0, xmmword ptr [reloc @RWD00]
       vpaddq   xmm0, xmm0, xmm0
       vpmovmskb eax, xmm0
						;; bbWeight=1    PerfScore 4.33

G_M58954_IG03:
       ret      
						;; bbWeight=1    PerfScore 1.00
RWD00  	dq	0000000000000001h, 0000000000000001h
; Total bytes of code: 20
```